packages/rpc-core/src/rpc-methods/__tests__/get-token-supply-test.ts

```diff
@@ -0,0 +1,60 @@
+import { Base58EncodedAddress } from '@solana/addresses';
+import { createHttpTransport, createJsonRpc } from '@solana/rpc-transport';
+import type { SolanaJsonRpcErrorCode } from '@solana/rpc-transport/dist/types/json-rpc-errors';
+import type { Rpc } from '@solana/rpc-transport/dist/types/json-rpc-types';
+import fetchMock from 'jest-fetch-mock-fork';
+
+import { Commitment } from '../common';
+import { createSolanaRpcApi, SolanaRpcMethods } from '../index';
+
+// See scripts/fixtures/spl-token-mint-account.json
+const tokenMintAddress = 'Gh9ZwEmdLJ8DscKNTkTqPbNwLNNBjuSzaG9Vp2KGtKJr' as Base58EncodedAddress;
+
+describe('getTokenSupply', () => {
+    let rpc: Rpc<SolanaRpcMethods>;
+    beforeEach(() => {
+        fetchMock.resetMocks();
+        fetchMock.dontMock();
+        rpc = createJsonRpc<SolanaRpcMethods>({
+            api: createSolanaRpcApi(),
+            transport: createHttpTransport({ url: 'http://127.0.0.1:8899' }),
+        });
+    });
+
+    (['confirmed', 'finalized', 'processed'] as Commitment[]).forEach(commitment => {
+        describe(`when called with \`${commitment}\` commitment`, () => {
+            it('returns total token supply', async () => {
+                expect.assertions(1);
+                const tokenAccountBalancePromise = rpc.getTokenSupply(tokenMintAddress, { commitment }).send();
+                await expect(tokenAccountBalancePromise).resolves.toMatchObject({
+                    context: {
+                        slot: expect.any(BigInt),
+                    },
+                    value: {
+                        amount: expect.any(String),
+                        decimals: expect.any(Number),
+                        uiAmount: expect.anything(), // Number or null
```

Comment:
Same suggestion to use `any(Number)` here

Reply:
Done !